### PR TITLE
ODC-7450: Complete Interval Dropdown for API calls

### PR DESCRIPTION
**Pull request by Aneesh-M-Bhat**
PR on top of #16 
Story: [ODC-7450](https://issues.redhat.com/browse/ODC-7450)
Purpose: 
Added intervals for Duration Card, Numbers Chart, StatusCard, TotalRunsCard

### CODE REVIEW

File: src/components/pipelines-overview/PipelineRunsTotalCard.tsx [hunk 1]
```diff
@@ -43,51 +43,44 @@ const PipelinesRunsTotalCard: React.FC<PipelinesRunsDurationProps> = ({
...
       data_type: DataType.PipelineRun,
       filter,
     })
       .then((response) => {
-        setRepoRun(response.summary[0].total);
+        setData(response.summary[0].total);
       })
       .catch((e) => {
-        console.error('Error in getSummary', e);
+        console.error('Unable to Post', e);
```
Comment (saumeya) [inline src/components/pipelines-overview/PipelineRunsTotalCard.tsx:57]: Nit: revert to Error in getSummary

File: src/components/pipelines-overview/utils.ts [hunk 2]
```diff
@@ -67,3 +67,18 @@ export const LAST_LANGUAGE_LOCAL_STORAGE_KEY = 'bridge/last-language';
 
 export const getLastLanguage = (): string =>
   localStorage.getItem(LAST_LANGUAGE_LOCAL_STORAGE_KEY);
+
+export const useInterval = (
```
Comment (saumeya) [inline src/components/pipelines-overview/utils.ts:71]: Where is the `No refresh` case handled?

**Review by saumeya — COMMENTED** [review]
Left minor comments, other looks good to me!

File: src/components/pipelines-overview/utils.ts (hunk 2 above)
Comment (Aneesh-M-Bhat) [reply src/components/pipelines-overview/utils.ts:71]: ya so, in the useInterval hook, when user selects no refresh it updates once, then in this custom hook i have put a condition to run setInterval() only if interval!==null, which is the value when you select no refresh.

File: src/components/pipelines-overview/PipelineRunsTotalCard.tsx (hunk 1 above)
Comment (Aneesh-M-Bhat) [reply src/components/pipelines-overview/PipelineRunsTotalCard.tsx:57]: ya had issue selecting 1 or the other, will revert to error in get summary.

**Review by saumeya — APPROVED** [review]
/lgtm

**Pull request closed without merging**